### PR TITLE
bsnes_libretro: various fixes

### PR DESCRIPTION
**Pull request by kwyxz**
Updated versioning, replaced SUPPLEMENTS with REQUIRES, updated revision
Also replaced $libDir with $addOnsDir

### CODE REVIEW

File: games-emulation/bsnes_libretro/bsnes_libretro-0.94.20171227git.recipe [hunk 1]
```diff
@@ -46,7 +44,7 @@ BUILD_PREREQUIRES="
 BUILD()
 {
 	for bsnesProfile in "accuracy" "balanced" "performance"; do
-		sed -e "s/@DISPLAY_VERSION@/v${portVersion%.git*}/" \
+		sed -e "s/@DISPLAY_VERSION@/v${portVersion%git}/" \
```
Comment (fbrosson) [inline games-emulation/bsnes_libretro/bsnes_libretro-0.94.20171227git.recipe]: We'll also have the same problem here as with fbalpha_libretro: `20171227git` < `git20171227` so `pkgman update` would think that `bsnet_libretro-20171227git` is older than `bsnet_libretro-git20171227`.
Users would have to manually uninstall `bsnet_libretro-git20171227`.

We can fix it by renaming the recipe to `bsnes_libretro-0.94_20171227.recipe` and the patchset to `bsnes_libretro-0.94_20171227.patchset`.
```diff
-		sed -e "s/@DISPLAY_VERSION@/v${portVersion%git}/" \
+		sed -e "s/@DISPLAY_VERSION@/v${portVersion/_/-}/" \
```

**Comment by kwyxz** [comment]
Fixed portVersion here too

**Comment by fbrosson** [comment]
Hmm, I just noticed we forgot to insert a `SOURCE_FILENAME`. Shall we do so with a `${portVersion/_/-}` ?
```diff
 CHECKSUM_SHA256="77255571b654ef1b7559608818393dc8377c657dc30aa7d3ec45e40662c99a31"
+SOURCE_FILENAME="bsnes-libretro-${portVersion/_/-}-$srcGitRev.tar.gz"
 SOURCE_DIR="bsnes-libretro-$srcGitRev"
```

**Comment by kwyxz** [comment]
Added SOURCE_FILENAME

File: games-emulation/bsnes_libretro/bsnes_libretro-0.94_20171227.recipe [hunk 2]
```diff
@@ -7,11 +7,12 @@ attempting to provide a reasonable balance between the previous two."
 HOMEPAGE="https://byuu.org/emulation/higan/"
 COPYRIGHT="2014-2018 byuu, the libretro team"
 LICENSE="GNU GPL v3"
-REVISION="1"
+REVISION="2"
 srcGitRev="4e572a3cfa10e56bedb6737b0494e8165f6490d5"
 SOURCE_URI="https://github.com/libretro/bsnes-libretro/archive/$srcGitRev.tar.gz"
 CHECKSUM_SHA256="77255571b654ef1b7559608818393dc8377c657dc30aa7d3ec45e40662c99a31"
 SOURCE_DIR="bsnes-libretro-$srcGitRev"
+SOURCE_FILENAME="bsnes-libretro-${portVersion/_/-}-$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline games-emulation/bsnes_libretro/bsnes_libretro-0.94_20171227.recipe]:
```diff
-SOURCE_DIR="bsnes-libretro-$srcGitRev"
 SOURCE_FILENAME="bsnes-libretro-${portVersion/_/-}-$srcGitRev.tar.gz"
+SOURCE_DIR="bsnes-libretro-$srcGitRev"
```
Yep, we prefer to put `SOURCE_DIR` after `SOURCE_FILENAME`. Thanks :)
Source: [HaikuPorter-Guidelines#ordering](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering)
(I did some metrics some time ago and found out that there are more recipes with `SOURCE_FILENAME` before `SOURCE_DIR` than the oposite, so we try to stick to the trend.)
(For recipe which already exist, we only swap if we also make a change near that line.)

**Comment by kwyxz** [comment]
Gotcha. Changed order.

**Pull request merged**